### PR TITLE
move graphql logic into integration_test_lib

### PR DESCRIPTION
**Pull request by QuiteStochastic**
https://app.zenhub.com/workspaces/velocity-62264fddc441a100183f7f86/issues/gh/minaprotocol/mina/13260

Previously, the only way for the test_executive to send graphql commands to nodes on the lucy testnet was to use the graphql functions in kubernetes_network.ml.  However kubenetes_network.ml is within the package `lib/integration_test_cloud_engine`, which would make it inaccessible to any other hypothetical engine which we may want to develop, therefore the new engine would have to reimplement it all from scratch.

Therefore I have ported the graphql functions and surrounding logic from `lib/integration_test_cloud_engine/kubernetes_network.ml` to `lib/integration_test_lib/graphql_requests.ml`

It was a pretty direct port, the only real dependency was that the old functions required that the node be passed to them.  That's really quite unnecessary, the functions only really need the Uri of the node, which is how the new functions work.

old:

```
val send_payment :
     logger:Logger.t
  -> t (* this is a Kubernetes_network.Node.t*)
  -> sender_pub_key:Signature_lib.Public_key.Compressed.t
  -> receiver_pub_key:Signature_lib.Public_key.Compressed.t
  -> amount:Currency.Amount.t
  -> fee:Currency.Fee.t
  -> signed_command_result Deferred.Or_error.t
```

new:

```
val send_payment :
     logger:Logger.t
  -> node_uri:Uri.t (* now we just pass in a common Uri.t type, which is a basic library type.  *)
  -> sender_pub_key:Signature_lib.Public_key.Compressed.t
  -> receiver_pub_key:Signature_lib.Public_key.Compressed.t
  -> amount:Currency.Amount.t
  -> fee:Currency.Fee.t
  -> signed_command_result Deferred.Or_error.t
```

### CODE REVIEW

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

File: src/app/test_executive/payments_test.ml [hunk 1]
```diff
@@ -78,7 +78,7 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
     }
 
   let run network t =
-    let open Network in
+    (* let open Network in *)
```
Comment (mrmr1993) [inline src/app/test_executive/payments_test.ml:81]: Delete

File: src/lib/integration_test_lib/intf.ml [hunk 2]
```diff
@@ -48,139 +48,141 @@ module Engine = struct
...
-        -> t
-        -> best_chain_block list Async_kernel.Deferred.Or_error.t
-
-      val must_get_best_chain :
-           ?max_length:int
-        -> logger:Logger.t
-        -> t
-        -> best_chain_block list Malleable_error.t
+      val get_ingress_uri : t -> Uri.t
+      (*
```
Comment (mrmr1993) [inline src/lib/integration_test_lib/intf.ml]: Delete commented code

File: src/lib/integration_test_lib/intf.ml [hunk 3]
```diff
@@ -192,9 +194,9 @@ module Engine = struct
 
       val dump_precomputed_blocks :
         logger:Logger.t -> t -> unit Malleable_error.t
-
-      val get_metrics :
-        logger:Logger.t -> t -> metrics_t Async_kernel.Deferred.Or_error.t
+      (*
```
Comment (mrmr1993) [inline src/lib/integration_test_lib/intf.ml]: Ditto

**Review by mrmr1993 — COMMENTED** [review]
Lgtm but see nits

**Review by mrmr1993 — APPROVED** [review]

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Pull request merged**